File: src/test/java/problems/easy/ClimbingStairs/ClimbingStairsTest.cs
// <copyright file="ClimbingStairsTest.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace Problems.Easy.ClimbingStairs
{
    using System.Collections.Generic;
    using NUnit.Allure.Core;
    using NUnit.Framework;

    using NUnit.Allure.Attributes;
    using NUnit.Allure.Core;
    using NUnit.Framework;

    [TestFixture]
    [AllureNUnit]
    [AllureEpic("epic Problems")]
    [AllureFeature("feature Easy")]
    [AllureStory("story Climbing stairs")]
    [AllureSuite("suite C#")]
    [AllureTag("tag C#")]
    public class ClimbingStairsTest
    {
        private Solution cut;

        public static IEnumerable<TestCaseData> GetInputData()
        {
            yield return new TestCaseData(1, 1);
            yield return new TestCaseData(2, 2);
            yield return new TestCaseData(3, 3);
            yield return new TestCaseData(4, 5);
            yield return new TestCaseData(5, 8);
        }

        [SetUp]
        public void SetUp()
        {
            this.cut = new Solution();
        }

        [Test]
        [TestCaseSource(nameof(GetInputData))]
        [AllureName("test Climbing stairs")]
        public void ShouldCalculateStairs(int input, int expectedResult)
        {
            Assert.That(this.cut.ClimbStairs(input), Is.EqualTo(expectedResult));
        }
    }
}

File: src/test/java/problems/easy/ClimbingStairs/ClimbingStairsTest.java
package problems.easy.ClimbingStairs;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.params.provider.Arguments.of;

import io.qameta.allure.Epic;
import io.qameta.allure.Feature;
import io.qameta.allure.Story;
import java.util.stream.Stream;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Tags;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

@Tags({@Tag("tag Java")})
@Epic("epic Problems")
@Feature("feature Easy")
@Story("story Climbing stairs")
@DisplayName("suite Java")
class ClimbingStairsTest {
    @ParameterizedTest
    @MethodSource("getInputData")
    @Tags({@Tag("tag Java")})
    @DisplayName("test Climbing stairs")
    void testClimbingStairs(int input, int expectedResult) {
        assertEquals(expectedResult, new Solution().climbStairs(input));
    }

    static Stream<Arguments> getInputData() {
        return Stream.of(of(1, 1), of(2, 2), of(3, 3), of(4, 5), of(5, 8));
    }
}
